gateway/1.4/modules/ROOT/pages/flex-gateway-getting-started-kubernetes.adoc

```diff
@@ -0,0 +1,650 @@
+KubeDNS is running at https://34.30.50.119/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+Metrics-server is running at https://34.30.50.119/api/v1/namespaces/kube-system/services/https:metrics-server:/proxy
+----
++
+If the cluster is not up, the command can return a result similar to this one:
++
+----
+To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
+error: the server doesn't have a resource type "services"
+----
++
+For additional information, see <<check-cluster>>.
+
+[[upgrade-failed]]
+* Some issues can cause the Helm upgrade command to hang for up to 5 minutes (due to the `--wait` flag in the command) before producing the following timeout error:
++
+----
+Error: UPGRADE FAILED: timed out waiting for the condition
+----
++
+This error typically means that your cluster configuration is incorrect or that your cluster configuration tool does not support this process. See <<k8-cluster-reqs>>. One potential cause is the failure to generate a `Service` resource of the `LoadBalancer` type. You can troubleshoot further by running the following commands:
+
+** {empty}
++
+[source,kubectl]
+----
+kubectl -n gateway describe deployment
+----
+
+** {empty}
++
+[source,kubectl]
+----
+kubectl -n gateway describe service
+----
+
+//TODO: this was from Israel. Not clear what upgrade error and 
+//      why we would uninstall the ingress for the namespace. 
+//Israel: "If you receive an upgrade error when execute the command, you can  
+//         remove the ingress released chart or omit the released argument:"
+//
+//## Command to remove ingress chart released
+//helm -n <namespace> uninstall ingress
+//
+// TODO: WHEN WOULD WE OMIT -i --create-namespace --wait?
+//## Command without install ingress chart arguments
+// helm -n <namespace> upgrade \
+//ingress flex-gateway/flex-gateway \
+//--set-file registration.content=<registration_file>
+
+[[check-cluster]]
+== Check Your Cluster
+
+Before deploying Flex Gateway to your Kubernetes cluster in <<deploy-flex>>, you must make sure that your cluster is ready. To check, you can use `kubectl` to list and select (use) a cluster and to get a list of nodes that the cluster uses. If any steps in this section fail, the deployment process is likely to produce errors.
+
+To prevent potential timeout and connectivity errors when running procedures in this guide, your cluster must meet the <<k8-cluster-reqs>>. 
+
+. Run `kubectl config get-contexts` to list the clusters, for example:
++
+[source,kubectl]
+----
+kubectl config get-contexts
+----
++
+This command is particularly helpful if you have multiple Kubernetes clusters. You can use the command to get the name of the the correct cluster. For example, assume that you have a variety of clusters (`my-cluster`, `my-aks`, `my-gke`, `my-eks`). The command prints a list of your clusters. 
++
+The following _example_ shows a list of clusters produced by the command:
```

Comment:
```suggestion
The following example shows a list of clusters produced by the command:
```

Reply:
Kim says to use italics for emphasis sparingly.

Reply:
I need to emphasize so they don't copy&paste.